src/UX/UserChartBuilder.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+/*
+ * This file is part of the Starfleet Project.
+ *
+ * (c) Starfleet <msantostefano@jolicode.com>
+ *
+ * For the full copyright and license information,
+ * please view the LICENSE file that was distributed with this source code.
+ */
+
+namespace App\UX;
+
+use App\Entity\Submit;
+use Symfony\UX\Chartjs\Builder\ChartBuilderInterface;
+use Symfony\UX\Chartjs\Model\Chart;
+
+class UserChartBuilder
+{
+    public function __construct(
+        private ChartBuilderInterface $chartBuilder,
+    ) {
+    }
+
+    /**
+     * @param array<Submit> $pendingSubmits
+     * @param array<Submit> $acceptedSubmits
+     * @param array<Submit> $rejectedSubmits
+     * @param array<Submit> $upcomingSubmits
+     */
+    public function buildUserChart(array $pastSubmits, array $futureSubmits, array $pastConferences, array $futureConferences): Chart
+    {
+        $chart = $this->chartBuilder->createChart(Chart::TYPE_DOUGHNUT);
+        $chart->setData([
+            'labels' => [
+                sprintf('%d talks given', \count($pastSubmits)),
+                sprintf('%d talks you will give soon', \count($futureSubmits)),
+                sprintf('%d conferences you went to', \count($pastConferences)),
+                sprintf('%d conferences you will attend soon', \count($futureConferences)),
+            ],
+            'datasets' => [
+                [
+                    'label' => 'Submitted talks',
```

Comment:
Please use the Translator

Reply:
and everywhere it should be used ;) (no hard coded text)

Reply:
Will be done in a future PR when we upgrade to SF 5.3, so I'm resolving for now while leaving as is.